target-i386/trace_helper.c

```diff
@@ -184,3 +169,72 @@ void HELPER(trace_st)(CPUArchState *env, target_ulong val, target_ulong addr)
 
     qemu_trace_add_operand(oi, 0x2);
 }
+
+
+OperandInfo * store_flag(short val, const char *reg_name)
+{
+    RegOperand * ro = g_new(RegOperand, 1);
+    reg_operand__init(ro);
+        
+    ro->name = g_strdup(reg_name);
+    strcpy(ro->name, reg_name);
+
+    OperandInfoSpecific *ois = g_new(OperandInfoSpecific, 1);
+    operand_info_specific__init(ois);
+    ois->reg_operand = ro;
+
+    OperandUsage *ou = g_new(OperandUsage, 1);
+    operand_usage__init(ou);
+    ou->written = 1;
+    OperandInfo *oi = g_new(OperandInfo, 1);
+    operand_info__init(oi);
+    oi->bit_length = 1;
+    oi->operand_info_specific = ois;
+    oi->operand_usage = ou;
+    oi->value.len = 1;
```

Comment:
your value is actually a `short` that is usually `16` bits of information encoded in a host specific way. On bigendian platforms you will always store `0`, as the most significant byte of a `1` stored in the `short int` is always `0`. You should always copy `sizeof(value)` into the data. Also, consider to use `_Bool`, or `bool` from `stdbool.h`